### PR TITLE
[New Model] Production Tracking Response Data Model

### LINKED ISSUE #547: [New Model]: ProvideProductionTrackingData
## Model Description

The Shop Floor Information Service will be extended in 2405 by a method to exchange process data collected during the production of a process for tracebility or documentation. This service will have two actions: a request asking for the data and the answer with the requested data.

This Request corresponds to the ProvideProductionTrackingData Reply:
## Relevant Standards

None

## Example Data

|Level|Property|mand./opt.|Semantic ID|
|--|--|--|--|
|CX-Header|recipientBpn|mandatory|string|
|CX-Header|context|mandatory|string|
|CX-Header|messageId|mandatory|UUID|
|CX-Header|sentDateTime|mandatory|string|
|CX-Header|version|mandatory|string|
|CX-Header|relatedMessageId|optional|UUID|
|SIS-Payload|versionDataModel|mandatory|string|
|SIS-Payload|serialNumber||
|SIS-Payload|processStepIdentifierList|1|
|SIS-Payload|  processStepId|1..n|
|SIS-Payload|  processStepId|1..n|
|SIS-Payload|       processParameterName|1..n|string|
||       (classification) / semanticID|1..n|SemanticID|
|SIS-Payload|       processParameterValue|1..n|string|
|SIS-Payload|       processParameterQuality|1..n|enum||Value: ok, inexact, noValue|              

## MS1 Criteria

- [x] The proposed aspect model does not exist already
- [x] The proposed aspect model does not extend an existing aspect but introduces completely new functionality
- [x] The proposal references a Catena-X use case
- [x] Relevant standards are mentioned/linked
- [x] A first draft of the data structure is provided in an example
- [x] A modeling steward has been assigned

### PR DESCRIPTION
**Pull request by FlorianDue**
## Description

Data Model for the response of the production tracking within the Shopfloor Information Service

 -->

Closes #547 

## MS2 Criteria
(to be filled out by PR reviewer)
- [x] the model **validates** with the SAMM SDS SDK in the version specified in the Readme.md of this repository by the time of the MS2 check  (e.g., 'java -jar samm-cli.jar aspect \<path-to-aspect-model\> validate ). The  SAMM CLI is available [here](https://eclipse-esmf.github.io/esmf-developer-guide/tooling-guide/samm-cli.html) and in [GitHub](https://github.com/eclipse-esmf/esmf-sdk/releases/tag/v2.3.2)
- [x] use **Camel-Case** (e.g., "MyModelElement" or "TimeDifferenceGmtId", when in doubt follow https://google.github.io/styleguide/javaguide.html#s5.3-camel-case)
- [x] the identifiers for all model elements **start with a capital letter** except for properties
- [x] the identifier for **properties starts with a small letter**
- [x] all model elements **at least contain the fields "preferred name" and "description"** in English language. The description must be comprehensible. It is not required to write full sentences but style should be consistent over the whole model
- [x] Property and the referenced Characteristic should not have the same name
- [x] the versioning in the URN **follows semantic versioning**, where minor version bumps are backwards compatible and major version bumps are not backwards compatible. 
- [x] use **abbreviations only when necessary** and if these are sufficiently common
- [x] **avoid redundant prefixes in property names** (consider adding properties to an enclosing Entity or even adapt the namespace of the model elements, e.g., instead of having two properties `DismantlerId` and `DismantlerName` use an Entity `Dismantler` with the properties `name` and `id` or use a URN like `io.catenax.dismantler:0.0.1`)
- [x] fields `preferredName` and `description` are not the same
- [x] **`preferredName` should be human readable** and follow normal orthography (e.g., no camel case but normal word separation)
- [x] name of aspect is singular except if it only has one property which is a Collection, List or Set. In theses cases, the aspect name is plural.
- [x] units are referenced from the SAMM unit catalog whenever possible
- [x] **use constraints** to make known constraints from the use case explicit in the aspect model 
- [x] when relying on **external standards**, they are referenced through a **"see"** element
- [x] all properties with an [simple type](https://eclipse-esmf.github.io/samm-specification/2.0.0/datatypes.html) have an example value
- [x] metadata.json exists with status "release"
- [x] generated json schema validates against example json payload
- [x] file RELEASE_NOTES.md exists and contains entries for proposed model changes 
- [x] all contributors to this model are mentioned in copyright header of model file

## MS3 Criteria
(to be filled out by semantic modeling team before merge to main-branch)
- [x] All required reviewers have approved this PR (see reviewers section)
- [x] The new aspect (version) will be implemented by at least one data provider
- [x] The new aspect (version) will be consumed by at least one data consumer
- [x] There exists valid test data
- [x] In case of a new (incompatible) major version to an existing version, a migration strategy has been developed
- [x] The model has at least version '1.0.0'
- [x] If a previous model exists, model deprecation has been checked for previous model
- [x] The release date in the Release Note is set to the date of the MS3 approval

### CODE REVIEW

File: io.catenax.shopfloor_information.provide_production_tracking/1.0.0/ProvideProductionTrackingData.ttl [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+@prefix ext-header2: <urn:samm:io.catenax.shared.message_header:2.0.0#>.
+
+:ProvideProductionTrackingData a samm:Aspect;
+    samm:preferredName "ProvideProductionTrackingData"@en;
+    samm:properties (:processStepIdentifierList ext-part:localIdentifiers ext-header2:header ext-header2:version);
+    samm:operations ();
+    samm:events ().
+
+:processStepIdentifierList a samm:Property;
+    samm:preferredName "processStepIdentifierList"@en;
```
Comment (agg3fe) [inline io.catenax.shopfloor_information.provide_production_tracking/1.0.0/ProvideProductionTrackingData.ttl:35]: preferredName value should be in plain English form without any camel case.
please correct all other places also.

File: io.catenax.shopfloor_information.provide_production_tracking/1.0.0/ProvideProductionTrackingData.ttl [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+:processStepId a samm:Property;
+    samm:preferredName "processStepId"@en;
+    samm:description "Id of a concrete process step"@en;
+    samm:characteristic :StringCharacteristic;
+    samm:exampleValue "step_id_12345".
+
+:processParameterValueList a samm:Property;
+    samm:preferredName "processParameterValueList"@en;
+    samm:description "A set of process parameter to the corresponsing process step"@en;
```
Comment (agg3fe) [inline io.catenax.shopfloor_information.provide_production_tracking/1.0.0/ProvideProductionTrackingData.ttl:62]: typo `corresponsing`

File: io.catenax.shopfloor_information.provide_production_tracking/1.0.0/ProvideProductionTrackingData.ttl [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+:semanticId a samm:Property;
+    samm:preferredName "semanticId"@en;
+    samm:description "reference to a semantic that characterizes the process parameter"@en;
+    samm:characteristic :StringCharacteristic;
+    samm:exampleValue "EClass123".
+
+:processParameterValue a samm:Property;
+    samm:preferredName "processParameterValue"@en;
+    samm:description "the concrete value of a process parameter, expressed as json string. Can only caontain a value if processParameterQuality is not NoValue"@en;
```
Comment (agg3fe) [inline io.catenax.shopfloor_information.provide_production_tracking/1.0.0/ProvideProductionTrackingData.ttl:89]: typo `caontain`

File: io.catenax.shopfloor_information.provide_production_tracking/1.0.0/ProvideProductionTrackingData.ttl [hunk 4]
```diff
@@ -0,0 +1,103 @@
+##########################################################################################
+# Copyright (c) 2023-2024 Fraunhofer-Gesellschaft zur Foerderung der angewandten Forschung e.V. (represented by Fraunhofer IOSB & Fraunhofer IWU & Fraunhofer IPA)
```
Comment (agg3fe) [inline io.catenax.shopfloor_information.provide_production_tracking/1.0.0/ProvideProductionTrackingData.ttl:2]: only 2024 is required

**Review by agg3fe — CHANGES_REQUESTED** [review]
please check my comments

File: io.catenax.shopfloor_information.provide_production_tracking/1.0.0/ProvideProductionTrackingData.ttl (hunk 4 above)
Comment (agg3fe) [reply io.catenax.shopfloor_information.provide_production_tracking/1.0.0/ProvideProductionTrackingData.ttl]: the issue is still there. please change the year to 2024 only.

File: io.catenax.shopfloor_information.provide_production_tracking/1.0.0/ProvideProductionTracking.ttl [hunk 5]
```diff
@@ -0,0 +1,137 @@
+##########################################################################################
+# Copyright (c) 2023-2024 Fraunhofer-Gesellschaft zur Foerderung der angewandten Forschung e.V. (represented by Fraunhofer IOSB & Fraunhofer IWU & Fraunhofer IPA)
```
Comment (agg3fe) [inline io.catenax.shopfloor_information.provide_production_tracking/1.0.0/ProvideProductionTracking.ttl:2]: 2024 without 2023

Comment (FlorianDue) [reply io.catenax.shopfloor_information.provide_production_tracking/1.0.0/ProvideProductionTracking.ttl]: corrected

**Review by agg3fe — APPROVED** [review]
LGTM
